backend/src/main/java/com/bakdata/conquery/util/search/solr/FilterValueIndexer.java

```diff
@@ -0,0 +1,231 @@
+
+		// Set rows to 0 because we don't want actual results, we are only interested in the total number
+		query.setRows(0);
+
+		try {
+			QueryResponse response = solrClient.query(query);
+			return response.getResults().getNumFound();
+		}
+		catch (SolrServerException | IOException e) {
+			throw new RuntimeException(e);
+		}
+	}
+
+	/**
+	 * @implNote This is not used, because this indexer only imports data and is not used by a {@link InternToExternMapper} to retrieve
+	 * the data. Data retrieval is done by {@link FilterValueSearch} which combines multiple {@link Searchable}s of a
+	 * {@link Filter}.
+	 */
+	@Override
+	public List<FrontendValue> findExact(String searchTerm, int maxValue) {
+		throw new UnsupportedOperationException("Must not be used. Use " + FilterValueSearch.class);
+	}
+	@Override
+	public void addItem(FrontendValue feValue, List<String> _keywords) {
+
+		SolrFrontendValue solrFrontendValue = new SolrFrontendValue(searchable, sourcePriority, feValue);
+
+		insertIntoChunk(solrFrontendValue);
+
+	}
+
+	private void insertIntoChunk(SolrFrontendValue solrFrontendValue) {
+
+		if (solrFrontendValue.value_s.isEmpty()) {
+			if (seenEmpty) {
+				log.trace("Skip indexing of {} for {}, because its 'value' is empty and was already added.", solrFrontendValue, searchable);
+				return;
+			}
+			seenEmpty = true;
+
+		}
+
+		if (!seenValues.add(solrFrontendValue.value_s)) {
+			log.trace("Skip indexing of {} for {}, because its 'value' has already been submitted to solr.", solrFrontendValue, searchable);
+			return;
+		}
+
+		openChunk.add(solrFrontendValue);
+
+		if (openChunk.size() >= updateChunkSize) {
+			openChunks.add(openChunk);
+			openChunk = new ArrayList<>(updateChunkSize);
+		}
+	}
+
+	private synchronized CompletableFuture<Void> submitChunk() {
+		submitter =  submitter.thenRunAsync(
+				() -> {
+					if (openChunks.isEmpty()) {
+						return;
+					}
+
+					// We chunk here for performance.
+					// Too many small document request cause a lot of overhead.
+					// A too large chunk slows request submission and solr.
+					do  {
+						int chunksCount = openChunks.size();
+
+						Collection<SolrFrontendValue> chunk = openChunks.poll();
+						log.debug("Adding {} (of currently ca. {}) documents for {}", chunk.size(), chunksCount*updateChunkSize, searchable);
+						registerValues(chunk);
+					} while (!openChunks.isEmpty());
+				},
+				executor
+		);
+
+		return submitter;
+	}
+
+	public void registerValuesRaw(Collection<String> values) {
+		// Convert values and prepare chunks
+		values.stream()
+			   .filter(Objects::nonNull)
+			   .filter(Predicate.not(String::isBlank))
+			   .map(value -> new SolrFrontendValue(searchable, sourcePriority, value, null, null))
+			   .forEach(this::insertIntoChunk);
+
+		// Check if there is already a submitter working on pending docs
+		if (submitter.isDone()) {
+			// Start submitting so we can start free pending documents
+			submitChunk();
+		}
```

Comment:
Ich glaube ich würde den submitter in seine eigene Klasse stecken? Das macht das queuing und den entsprechenden state etwas lesbarer